dartagnan/src/main/java/com/dat3m/dartagnan/GlobalSettings.java

```diff
@@ -24,8 +25,15 @@ private GlobalSettings() {}
             secure = true)
     private int arch_precision = -1;
 
+    @Option(name = MIXED_TYPE,
+            description = "If false, all non-boolean, non-aggregate data types are unified into one common integer type determined by " + ARCH_PRECISION + ".",
+            secure = true)
+    private boolean mixedType = false;
```

Comment:
We should definitely change the options. 
I would suggest to use `BV` by default with 64-bit precision for pointers and proper program types.
We can then have a single `useIntegers` option that makes everything integer.
We can get rid of `arch_precision` because everything apart from `64` is wrong for LLVM code (which often assumes pointers to be 64 bit) and the integer setting "`-1`" is better expressed by a separate option

Reply:
We can just change our configuration for the unit tests to still use integers! No need to compromise the performance there.

Reply:
These changes are not yet implemented

Reply:
They are implemented now.  A proper pointer type is planned for another PR.

Reply:
Looks like `AbstractCompilationTest` etc. still need mathematical integers for pointers.  Maybe I should revert the removal of `modeling.precision` for now and have it properly handled later.

Reply:
The current patch has `lastValue` return 64-bit vectors if `encoding.integers` is false.